src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -354,6 +392,10 @@ class DeltaCatalog(val spark: SparkSession) extends DelegatingCatalogExtension
           throw new IllegalArgumentException(s"Can't set location multiple times. Found " +
             s"${locations.asInstanceOf[Seq[SetProperty]].map(_.value())}")
         }
+        if (isPathIdentifier(ident)) {
+          // Doesn't make sense to set the location of a path based table
+          throw new AnalysisException("Database 'delta' not found")
```

Comment:
I think we should throw an error saying it doesn't make sense rather than pretending to be a database.

Reply:
this was the old failure message. Analysis would fail, therefore it would throw this exception. I'm happy to throw a better error message (and throw a better message with the old code path as well).